File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        answer = head
        current = head
        previous= None
        flag = True
        
        while current != None:
            if current.next == None:
                break

            swap1 = current
            swap2 = current.next
            current = swap2.next
            if previous != None:
                previous.next = swap2
            previous = swap1
            if flag:
                answer = swap2
                flag = False

            swap1.next = swap2.next
            swap2.next = swap1

        

        return answer

File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapPairs(ListNode head) {
        ListNode newHead = new ListNode(0, head);
        ListNode prev = newHead;
        ListNode curr = head;

        while (curr != null && curr.next != null){
            ListNode swap1 = curr;
            ListNode swap2 = curr.next;

            curr = swap2.next;
            swap2.next = swap1;
            swap1.next = curr;

            prev.next = swap2;
            prev = swap1;
        }
        
        return newHead.next;
    }

}